pkg/imgutil/commit/commit.go

```diff
@@ -173,6 +191,34 @@ func Commit(ctx context.Context, client *containerd.Client, container containerd
 	return configDigest, nil
 }
 
+// ensureManifestLayers checks layers specified in manifest, reconstruct contents if it is missing and change manifest
+func ensureManifestLayers(ctx context.Context, manifest *ocispec.Manifest, cs content.Store, config *ocispec.Image, sn snapshots.Snapshotter, comparer diff.Comparer) error {
+	if len(manifest.Layers)+1 != len(config.RootFS.DiffIDs) {
+		return fmt.Errorf("layer size in manifest(%v)+1 does not match with config(%v)", len(manifest.Layers), len(config.RootFS.DiffIDs))
+	}
+	for i, layer := range manifest.Layers {
+		_, err := cs.Info(ctx, layer.Digest)
+		if err == nil {
+			continue
+		}
+		if !errdefs.IsNotFound(err) {
+			return fmt.Errorf("get layer %v failed: %w", layer.Digest.String(), err)
+		}
+		snapshotID := identity.ChainID(config.RootFS.DiffIDs[:i+1]).String()
+		logrus.Infof("creating layer chainid:%v config diffid:%v\n", snapshotID, config.RootFS.DiffIDs[i+1])
+		diffLayerDesc, diffID, err := createDiff(ctx, snapshotID, sn, cs, comparer)
+		if err != nil {
+			return fmt.Errorf("failed to compress layer for %v: %w", layer.Digest.String(), err)
+		}
+		if diffID != config.RootFS.DiffIDs[i] {
+			logrus.Infof("newly created Diff id %v does not match with diff id in config %v: %v", diffID, config.RootFS.DiffIDs[i], err)
```

Comment:
This should be a warning. Also, it should be warned that the contents of the image has been changed by this diff computation.

Reply:
This func does not change content of the original image, it reconstructs the content for new image if original content is missing. So if nothing i changed for original image consumers.